core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -288,9 +288,7 @@ static void addBindPermissions(Permissions policy, Settings settings) throws IOE
         for (Map.Entry<String, Settings> entry : profiles.entrySet()) {
             Settings profileSettings = entry.getValue();
             String name = entry.getKey();
-            String transportRange = profileSettings.get("port",
-                                        settings.get("transport.tcp.port",
-                                                NettyTransport.DEFAULT_PORT_RANGE));
+            String transportRange = profileSettings.get("port",NettyTransport.PORT.get(settings));
```

Comment:
That's an excellent point. I will undo this, but looking at the class I think there is more work to do as other changes does similar stuff ( I can do that). Also, for my understanding - what is the difference in terms of class loading between running that static code (NettyTransport.PORT) and accessing a final static `NettyTrasnport.DEFAULT_PORT_RANGE` that was before?

Last - if there is any way to determine the list of classes loaded at the point security kicks in, I wonder if we should add an assertion on that with a limitted set of white listed classes? This is so easy to slip without anyone noticing.